### PR TITLE
fix clickhouse_driver inserts

### PR DESCRIPTION
**Pull request by CheSema**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 0a85fc6524788692ee5ddfe8c0cdd4a2b53dfee3 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/72813/0a85fc6524788692ee5ddfe8c0cdd4a2b53dfee3/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/b364af07f9253814bd808cae88822aed6a3a2d0b/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Server/TCPHandler.cpp [hunk 1]
```diff
@@ -37,6 +37,7 @@
 #include <Poco/Net/NetException.h>
 #include <Poco/Net/SocketAddress.h>
 #include <Poco/Util/LayeredConfiguration.h>
+#include "Common/StackTrace.h"
```
Comment (Algunenano) [inline src/Server/TCPHandler.cpp:40]: Doesn't look necessary. If it is, then please use <>

File: src/Server/TCPHandler.cpp [hunk 2]
```diff
@@ -1110,8 +1111,13 @@ void TCPHandler::processInsertQuery(QueryState & state)
                 startInsertQuery(state);
 
             while (receivePacketsExpectDataConcurrentWithExecutor(state))
+            {
                 executor.push(std::move(state.block_for_insert));
 
+                sendLogs(state);
```
Comment (Algunenano) [inline src/Server/TCPHandler.cpp:1117]: Should we be sending progress too? It'd be nice to leave a comment explaining why we send some values and not others

**Review by Algunenano — COMMENTED** [review]
Left some questions. As this was present in 24.11 it should be changed to BugFix and backported there

File: src/Server/TCPHandler.cpp (hunk 2 above)
Comment (CheSema) [reply src/Server/TCPHandler.cpp:1117]: To be honest -- 'I do not know'
It was and it is still overcomplicated question what and when we send something.
I has checked the previous version of that code: 
progress has not been sent for each block when `processInsertQuery`

Lest try to add `sendProgress` and check the tests. It makes sence.

Comment (CheSema) [reply src/Server/TCPHandler.cpp:1118]: No. If I add `sendProgress` tests fail.

**Review by Algunenano — APPROVED** [review]

**Pull request merged**